my_each.rb

```diff
@@ -1,6 +1,143 @@
 module Enumerable
-    def my_each(array)
-      for i in 0...array.length
-        p "gaa"
+  def my_each
+    (0...length).each do |i|
+      yield self[i]
     end
-end
\ No newline at end of file
+    self
+  end
+
+  def my_each_with_index
+    (0...length).each do |i|
+      yield self[i], i
+    end
+    self
+  end
+
+  def my_select
+    new_array = []
+    my_each do |element|
+      new_array.push(element) if yield element
+    end
+    new_array
+  end
+
+  def my_all?
+    result = true
+    my_each do |element|
+      statement = if block_given?
+                    yield element
+                  else
+                    element
+                  end
+
+      unless statement
+        result = false
+        break
+      end
+
+      result = true
+    end
+    result
+  end
+
+  def my_any?
+    result = true
+    my_each do |element|
+      statement = if block_given?
+                    yield element
+                  else
+                    element
+                  end
+
+      if statement
+        result = true
+        break
+      end
+      result = false
+    end
+    result
+  end
```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method doesn't consider the possibility when a class is given as an argument `(some_enumerable.my_any?(String)).`

- Your method doesn't consider the possibility when a Regex is given as an argument `(some_enumerable.my_any?(/w/)).`

- Your method doesn't consider the possibility when a string is given as an argument `(some_enumerable.my_any?('cow')).`